tests/test_test_dataframe.py

```diff
@@ -35,10 +37,13 @@ def spark(tmp_path_factory) -> SparkSession:
             "spark.jars.packages",
             "org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.5,org.apache.spark:spark-avro_2.12:3.5.5",
         )
+        .config('spark.driver.host', '127.0.0.1')
         .getOrCreate()
     )
     spark.sparkContext.setLogLevel("WARN")
     print(f"Using PySpark version {spark.version}")
+
```

Comment:
AFAIK this can be done a bit differently, no need to take request explicitly
```python
  yield spark
  spark.stop()
```
but on session scope and local master idk if that will have much difference, spark should be stopped with test process